price-estimator/src/main.rs

```diff
@@ -64,22 +70,45 @@ fn main() {
     let web3 = web3_provider(&http_factory, options.node_url.as_str(), options.timeout).unwrap();
     // The private key is not actually used but StableXContractImpl requires it.
     let private_key = PrivateKey::from_raw([1u8; 32]).unwrap();
-    let _contract = Arc::new(
+    let contract = Arc::new(
         StableXContractImpl::new(&web3, private_key, 0)
             .wait()
             .unwrap(),
     );
 
-    // TODO: create event based orderbook
-    // TODO: handle http requests
+    let orderbook = EventBasedOrderbook::new(contract, web3, options.orderbook_file);
+    orderbook.initialize().wait().unwrap();
+    let orderbook = Arc::new(Orderbook::new(orderbook));
 
     let mut runtime = runtime::Builder::new()
         .threaded_scheduler()
         .enable_all()
         .build()
         .unwrap();
 
-    runtime.block_on(warp::serve(warp::any().map(|| "")).run(([127, 0, 0, 1], 8080)));
+    let orderbook_task = runtime.spawn(update_orderbook_forever(
+        orderbook,
+        options.orderbook_update_interval,
+    ));
+
+    // TODO: handle http requests
```

Comment:
Is this to be done here in this PR or another?

Reply:
Another